flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -613,21 +617,42 @@ public void killCluster(ApplicationId applicationId) throws FlinkException {
                         yarnApplication,
                         validClusterSpecification);
 
-        // print the application id for user to cancel themselves.
-        if (detached) {
-            final ApplicationId yarnApplicationId = report.getApplicationId();
-            logDetachedClusterInformation(yarnApplicationId, LOG);
-        }
+        final ApplicationId yarnApplicationId = report.getApplicationId();
+        applicationId = yarnApplicationId;
+
+        AsyncClusterClientProvider<ApplicationId> clusterClientProvider =
```

Comment:
I don't think that we should create annonymous inner classes. This gives access to implementation details we should not expose. E.g. what if the `flinkConfiguration` changes? I think it would be better to properly separate the `AsyncClusterClientProvider` implementation from the `YarnClusterDescriptor`.

Reply:
Got it. Now the latest commit introduce the `AsyncClusterClientProvider` that extends the `ClusterClientProvider`